library/std/src/backtrace.rs

```diff
@@ -353,6 +364,14 @@ impl Backtrace {
     }
 }
 
+impl<'a> Backtrace {
+    /// Returns an iterator over the backtrace frames.
+    #[unstable(feature = "backtrace_frames", issue = "79676")]
+    pub fn frames(&'a self) -> &'a [BacktraceFrame] {
```

Comment:
Wouldn't it be better if this were an iterator? That could make lazily walking the stack cheaper at some point in the future.

E.g. Java introduced a [StackWalker](https://docs.oracle.com/javase/9/docs/api/java/lang/StackWalker.html) API to make partial stack traces cheaper.

Reply:
This was indeed discussed. There's some additional discussion and context that isn't apparent because it's in the previous iteration of this PR. Apologies for that. 

You can read some of that discussion [here](https://github.com/rust-lang/rust/pull/78299#discussion_r546175876).